src/quest_manager/tests/test_views.py

```diff
@@ -132,9 +133,27 @@ def setUp(self):
         self.quest1 = mommy.make(Quest)
         self.quest2 = mommy.make(Quest)
 
-        self.sub1 = mommy.make(QuestSubmission, user=self.test_student1, quest=self.quest1)
-        self.sub2 = mommy.make(QuestSubmission, quest=self.quest1)
-        self.sub3 = mommy.make(QuestSubmission, quest=self.quest2)
+        self.sub1 = mommy.make(
+            QuestSubmission,
+            user=self.test_student1,
+            quest=self.quest1,
+            is_completed=True,
+            semester=self.sem
+        )
+
+        self.sub2 = mommy.make(
+            QuestSubmission,
+            quest=self.quest1,
+            is_completed=True,
+            semester=self.sem
+        )
+
+        self.sub3 = mommy.make(
+            QuestSubmission,
+            quest=self.quest2,
+            semester=self.sem,
+            is_completed=False
+        )
```

Comment:
I don't think you should edit the general setup for one specific test?  Is this why you needed to delete all those other tests, because changing the setup causes the tests to fail?  In general, never mess with existing tests.  Just add, add, add.  There is rarely a need to REMOVE a working test.

Reply:
I changed the general setup coz `QuestSubmission.objects.all_completed(user)` doesn't return anything as it wasn't assigned to a semester and the quest wasn't complete. Although I've only used `self.sub1` in the test cases.

Should I revert it back and just add another QuestSubmission dummy data?

Reply:
Yes, I would prefer that.  Add your own test methods as needed and leave those that are already there and working (not broken)

Reply:
Also, don't change the setup, as all the other tests assume that data is set up the way it is.  It might not be a problem, but I'd rather not change working tests at all, just add to them with your own. and don't edit setup.  If you want a different setup, then make a new test class with your own setup.